lib/chef/provider/git.rb

```diff
@@ -301,7 +301,16 @@ def run_options(run_opts = {})
           # user who is executing `git` not the user running Chef.
           env["HOME"] = begin
             require "etc"
-            Etc.getpwnam(@new_resource.user).dir
+            case @new_resource.user
```

Comment:
I would probably change this to `when Integer; Etc.getpwuid(@new_resource.user).dir else Etc.getpwnam(@new_resource.user.to_s).dir` and trust in the resource to handle validation.